src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -45,10 +52,18 @@ public SubstringConstraint(string expected) : base(expected)
         /// <returns>True for success, false for failure</returns>
         protected override bool Matches(string actual)
         {
-            if (this.caseInsensitive)
-                return actual != null && actual.ToLower().IndexOf(expected.ToLower()) >= 0;
-            else
-                return actual != null && actual.IndexOf(expected) >= 0;
+            return actual != null && actual.IndexOf(expected, comparisonType) >= 0;
```

Comment:
`.ToLower()` is specifically the equivalent of `StringComparison.CurrentCultureIgnoreCase`. So the logic, without breaking changes, is:

`comparisonType ?? (caseInsensitive ? StringComparison.CurrentCultureIgnoreCase : StringComparison.Ordinal)`